File: leetcode3/최원준/3637. Trionic Array I.py
#

'''
1. 아이디어 :
2번 꺾이는지 확인합니다.

2. 시간복잡도 :
    O(n)

3. 자료구조/알고리즘 :


'''
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        n = len(nums)
        count = 0

        if nums[0] >= nums[1] or nums[n-2] >= nums[n-1]:
            return False

        for i in range(1, n-1):
            prev, curr, next = nums[i-1], nums[i], nums[i+1]
            if prev == curr or curr == next:
                return False

            if prev < curr and curr > next:
                count+=1
            elif prev > curr and curr < next:
                count+=1
        return count == 2

File: leetcode3/이진희/3637_Trionic Array I.java
/*

1. 아이디어 : 증가 -> 감소 -> 증가 순으로 반드시 배열이 유지되어야 함
              idx를 증가시키며 각각의 구간을 만족하는지 확인한다

2. 시간복잡도 : O(N)

3. 자료구조/알고리즘 : 완전탐색

 */

class Solution {
    public boolean isTrionic(int[] nums) {
        
        int idx = 0;

        while(idx < nums.length-1 && nums[idx]<nums[idx+1]) idx++;

        if(idx == 0) return false;

        while(idx < nums.length-1 && nums[idx]>nums[idx+1]) idx++;

        if(idx == nums.length-1) return false;

        while(idx < nums.length-1 && nums[idx]<nums[idx+1]) idx++;

        if(idx == nums.length-1) return true;
        else return false;
    }
}